crates/c-api/include/wasmtime/val.h

```diff
@@ -119,24 +119,38 @@ typedef uint8_t wasmtime_v128[16];
  */
 typedef union wasmtime_valunion {
   /// Field used if #wasmtime_val_t::kind is #WASMTIME_I32
+  ///
+  /// Note that this field is always stored in a little-endian format.
```

Comment:
Maybe store `char[4]`, `char[8]`, ... instead? Otherwise people will likely forget to do the endianness transform. Especially as all common architectures are little endian too.

Reply:
You need funky casts either way to support big endian systems.